### PR TITLE
Introduction of Laundry device support

### PR DESCRIPTION
**Pull request by warrenrees**
Contains GE Washer support and preliminary Dryer support

### CODE REVIEW

File: gekitchensdk/erd/erd_configuration.py [hunk 1]
```diff
@@ -113,5 +113,22 @@ def erd_encode(self, value: Any) -> str:
...
+    ErdConfigurationEntry(ErdCode.DISHWASHER_RINSE_AGENT, ErdRinseAgentConverter(), ErdCodeClass.DISHWASHER_SENSOR),
+
+    # Laundry
+    ErdConfigurationEntry(ErdCode.LAUNDRY_MACHINE_STATE, MachineStateConverter(), ErdCodeClass.LAUNDRY_SENSOR),
+    ErdConfigurationEntry(ErdCode.LAUNDRY_MACHINE_SUBCYCLE, MachineSubCycleConverter(), ErdCodeClass.LAUNDRY_SENSOR),
+    ErdConfigurationEntry(ErdCode.LAUNDRY_CYCLE, LaundryCycleConverter(), ErdCodeClass.LAUNDRY_SENSOR),
+    ErdConfigurationEntry(ErdCode.LAUNDRY_DOOR, LaundryDoorStatusConverter(), ErdCodeClass.DOOR),
+    ErdConfigurationEntry(ErdCode.LAUNDRY_DOOR_LOCK, LaundryDoorLockStatusConverter(), ErdCodeClass.LOCK_CONTROL),
+    ErdConfigurationEntry(ErdCode.LAUNDRY_REMOTE_STATUS, RemoteStatusConverter(), ErdCodeClass.LAUNDRY_SENSOR),
+    ErdConfigurationEntry(ErdCode.LAUNDRY_DELAY_TIME_REMAINING, ErdTimeRemainingConverter(), ErdCodeClass.LAUNDRY_SENSOR),
```
Comment (simbaja) [inline gekitchensdk/erd/erd_configuration.py:125]: Should be timer type?

File: gekitchensdk/erd/erd_configuration.py [hunk 2]
```diff
@@ -113,5 +113,22 @@ def erd_encode(self, value: Any) -> str:
...
-    ErdConfigurationEntry(ErdCode.DISHWASHER_RINSE_AGENT, ErdRinseAgentConverter(), ErdCodeClass.DISHWASHER_SENSOR)
+    ErdConfigurationEntry(ErdCode.DISHWASHER_RINSE_AGENT, ErdRinseAgentConverter(), ErdCodeClass.DISHWASHER_SENSOR),
+
+    # Laundry
+    ErdConfigurationEntry(ErdCode.LAUNDRY_MACHINE_STATE, MachineStateConverter(), ErdCodeClass.LAUNDRY_SENSOR),
+    ErdConfigurationEntry(ErdCode.LAUNDRY_MACHINE_SUBCYCLE, MachineSubCycleConverter(), ErdCodeClass.LAUNDRY_SENSOR),
+    ErdConfigurationEntry(ErdCode.LAUNDRY_CYCLE, LaundryCycleConverter(), ErdCodeClass.LAUNDRY_SENSOR),
+    ErdConfigurationEntry(ErdCode.LAUNDRY_DOOR, LaundryDoorStatusConverter(), ErdCodeClass.DOOR),
+    ErdConfigurationEntry(ErdCode.LAUNDRY_DOOR_LOCK, LaundryDoorLockStatusConverter(), ErdCodeClass.LOCK_CONTROL),
+    ErdConfigurationEntry(ErdCode.LAUNDRY_REMOTE_STATUS, RemoteStatusConverter(), ErdCodeClass.LAUNDRY_SENSOR),
```
Comment (simbaja) [inline gekitchensdk/erd/erd_configuration.py:124]: Probably should just be a boolean, see the oven for example

File: gekitchensdk/erd/erd_configuration.py [hunk 3]
```diff
@@ -113,5 +113,22 @@ def erd_encode(self, value: Any) -> str:
...
     ErdConfigurationEntry(ErdCode.DISHWASHER_OPERATING_MODE, OperatingModeConverter(), ErdCodeClass.DISHWASHER_SENSOR),
-    ErdConfigurationEntry(ErdCode.DISHWASHER_RINSE_AGENT, ErdRinseAgentConverter(), ErdCodeClass.DISHWASHER_SENSOR)
+    ErdConfigurationEntry(ErdCode.DISHWASHER_RINSE_AGENT, ErdRinseAgentConverter(), ErdCodeClass.DISHWASHER_SENSOR),
+
+    # Laundry
+    ErdConfigurationEntry(ErdCode.LAUNDRY_MACHINE_STATE, MachineStateConverter(), ErdCodeClass.LAUNDRY_SENSOR),
+    ErdConfigurationEntry(ErdCode.LAUNDRY_MACHINE_SUBCYCLE, MachineSubCycleConverter(), ErdCodeClass.LAUNDRY_SENSOR),
+    ErdConfigurationEntry(ErdCode.LAUNDRY_CYCLE, LaundryCycleConverter(), ErdCodeClass.LAUNDRY_SENSOR),
+    ErdConfigurationEntry(ErdCode.LAUNDRY_DOOR, LaundryDoorStatusConverter(), ErdCodeClass.DOOR),
+    ErdConfigurationEntry(ErdCode.LAUNDRY_DOOR_LOCK, LaundryDoorLockStatusConverter(), ErdCodeClass.LOCK_CONTROL),
```
Comment (simbaja) [inline gekitchensdk/erd/erd_configuration.py:123]: Perhaps this should be boolean?

File: gekitchensdk/erd/values/laundry/drynessnew_level.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import enum
```
Comment (simbaja) [inline gekitchensdk/erd/values/laundry/drynessnew_level.py:1]: DrynessNewLevel looks like DrynessLevel, they should probably be combined into the same type (you can then use the same converter for both ERD values)

File: gekitchensdk/erd/values/laundry/erd_cycle.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
+import enum
```
Comment (simbaja) [inline gekitchensdk/erd/values/laundry/erd_cycle.py:1]: This file looks duplicative of erd_laundry_cycle

**Review by simbaja — CHANGES_REQUESTED** [review]

File: gekitchensdk/erd/values/laundry/erd_cycle.py (hunk 5 above)
Comment (warrenrees) [reply gekitchensdk/erd/values/laundry/erd_cycle.py:1]: you're correct, this was a duplicate, i've deleted it.

File: gekitchensdk/erd/erd_configuration.py (hunk 1 above)
Comment (warrenrees) [reply gekitchensdk/erd/erd_configuration.py:125]: Good catch, updated.

File: gekitchensdk/erd/erd_configuration.py (hunk 2 above)
Comment (warrenrees) [reply gekitchensdk/erd/erd_configuration.py:124]: Updated

File: gekitchensdk/erd/erd_configuration.py (hunk 3 above)
Comment (warrenrees) [reply gekitchensdk/erd/erd_configuration.py:123]: Also updated

File: gekitchensdk/erd/values/laundry/drynessnew_level.py (hunk 4 above)
Comment (warrenrees) [reply gekitchensdk/erd/values/laundry/drynessnew_level.py:1]: Another good call, updated.

Comment (warrenrees) [reply gekitchensdk/erd/values/laundry/drynessnew_level.py]: I've been mulling this over, while they both have the same items, the values differ (eg: DrynessLevel has a INVALID level = 5, while DrynessNewLevel has a INVALID value of 0, which pushes the remaining values.

I'll consolidate them into the one set of files however I believe we'll still need to have the separate converter, unless I pass an argument to the function specifying which MAP should be referenced.

File: gekitchensdk/erd/values/laundry/erd_drynessnew_level.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import enum
```
Comment (simbaja) [inline gekitchensdk/erd/values/laundry/erd_drynessnew_level.py:1]: I think you can remove this file based on the other changes made.

Comment (warrenrees) [reply gekitchensdk/erd/values/laundry/erd_drynessnew_level.py]: done

**Pull request merged**